File: lib/smache.cc
//
// smache.cc
//

#include <smache/smache.hh>

Smache::Smache(uint32_t _r, uint32_t _w, uint32_t _n, Chunker* _chunker) : r(_r), w(_w), n(_n), chunker(_chunker)
{
}

Smache::~Smache()
{
}

Hash Smache::dcreate(int fd, uint64_t offset, uint64_t length)
{
    return Hash();
}

Hash Smache::dappend(int fd, uint64_t length)
{
    return Hash();
}

bool Smache::dread(Hash key, int fd, uint64_t offset, uint64_t length)
{
    return false;
}

Hash Smache::dwrite(Hash key, int fd, uint64_t offset, uint64_t length)
{
    return Hash();
}

Hash Smache::dremove(Hash key, uint64_t offset, uint64_t length)
{
    return Hash();
}

Hash Smache::dtruncate(Hash key, uint64_t length)
{
    return Hash();
}

bool Smache::imap(const char* name, Hash key)
{
    return false;
}

bool Smache::iremap(const char* name, Hash key)
{
    return false;
}

bool Smache::iunmap(const char* name)
{
    return false;
}

File: python/smache/smache.py
#!/usr/bin/env python

import native

from chunks import *
from exceptions import *

class Smache(native.Smache):
    compression = False

    def __init__(self, r, w, n, cas, compression):
        native.Smache.__init__(self, r, w, n, cas)
        self.compression = compression        

    def info(self, h):
        return Chunk(self.dinfo(h))

    def create(self, fd, offset, length):
        return Hash(self.dcreate(fd, offset, length))
    
    def append(self):
        return Hash(self.dappend(fd, length))
    
    def read(self):
        return self.dread(h, fd, offset, length)
    
    def write(self):
        return self.dwrite(h, fd, offset, length)
    
    def remove(self, h):
        return Hash(self.dremove(h, offset, length))
    
    def truncate(self, h, length):
        return Hash(self.dtruncate(h, length))
    
    def map(self, n, h):
        return self.imap(n, h)

    def lookup(self, n):
        return Hash(self.ilookup(n))

    def remap(self, n, h):
        return self.iremap(n, h)

    def unmap(self, n):
        return self.iunmap(n)
